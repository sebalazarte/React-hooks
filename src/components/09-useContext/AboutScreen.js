import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const userContext = useContext(UserContext);
    const { user, setUser } = userContext;

    const handleClick = () => {
        setUser({});
    }

    return (
        <div>
            <h1>About Screen</h1>
            <hr />
            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>
            <button
                className="btn btn-outline-primary"
                type="button"
                onClick={handleClick}
            >Logout</button>
        </div>
    )
}
