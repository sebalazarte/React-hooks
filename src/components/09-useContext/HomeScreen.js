import React, {useContext} from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

    const userContext = useContext(UserContext);
    const {user} = userContext;

    return (
        <div>
            <h1>Home Screen</h1>
            <hr/>
            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>
            
        </div>
    )
}
