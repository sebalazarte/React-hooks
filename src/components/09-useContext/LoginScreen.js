import React, { useContext } from 'react';
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const userContext = useContext(UserContext);
    const { user, setUser } = userContext;

    const handleClick = () => {

        const newUser = {
            id: 123,
            name: 'Sebastian',
            email: 'sebalazarte@outlook.com'
        };

        setUser(newUser);
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button
                className="btn btn-outline-primary"
                type="button"
                onClick={handleClick}
            >Login</button>

            <pre className="blockquote">
                {JSON.stringify(user, null, 3)}
            </pre>

        </div>
    )
}
