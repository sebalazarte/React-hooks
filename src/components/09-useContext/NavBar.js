import React from 'react'
import { NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
          <NavLink className="navbar-brand" to="/">
            useContext
          </NavLink>
          <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarText" aria-controls="navbarText" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" />
           </button>
                <div className="collapse navbar-collapse" id="navbarText">
                  <ul className="navbar-nav mr-auto">
                    <li className="nav-item">
                      <NavLink 
                        exact activeClassName="active" 
                        className="nav-link" 
                        to="/">
                        Home
                      </NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink 
                            exact 
                            activeClassName="active" 
                            className="nav-link" 
                            to="/about">
                        About
                      </NavLink>
                    </li>
                  </ul>
                <NavLink 
                    exact 
                    activeClassName="active" 
                    className="btn btn-outline-info my-2 my-sm-0" 
                    to="/login">
                    Login
                  </NavLink>
                  <NavLink
                    exact 
                    activeClassName="active"
                    className="btn btn-outline-info my-2 my-sm-0 ml-3"
                    to="/register"
                  >
                    Register
                  </NavLink>
                </div>
              </nav>
        
    );
}
