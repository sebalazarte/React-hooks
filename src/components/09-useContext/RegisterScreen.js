import React from 'react'

export const RegisterScreen = () => {
    return (
        <div>
            <h1>Register screen</h1>
            <hr/>
        </div>
    )
}
