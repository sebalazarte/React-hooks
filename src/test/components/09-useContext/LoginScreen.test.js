import React from 'react';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { mount } from 'enzyme';

describe('LoginScreen', () => {
    

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe ejecutar el setUser con el argumento esperado', () => {
        
        const user = {
            id: 123,
            name: 'Sebastian',
            email: 'sebalazarte@outlook.com'
        };

        // wrapper.find('button').prop('onClick')(); //es lo mismo
        wrapper.find('button').simulate('click');
        expect(setUser).toBeCalledWith(user);
    });

});