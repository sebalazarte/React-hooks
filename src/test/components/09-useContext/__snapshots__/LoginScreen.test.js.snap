// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
  <pre
    className="blockquote"
  />
</div>
`;
